extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts

```diff
@@ -589,28 +589,49 @@ export class SelectStorageAccountDialog {
 		const containerName = this._blobContainer.name;
 		const templates = this.migrationStateModel._armTemplateResult.templates!;
 		const sharedKeyCredential = new StorageSharedKeyCredential(this._storageAccount.name, storageKeys.keyName1);
-
-		const sasToken = generateBlobSASQueryParameters({
-			containerName,
-			permissions: BlobSASPermissions.parse("racwd"),
-			expiresOn: new Date(new Date().valueOf() + 86400),
-		},
+		var deployToAzureUrl: string = "";
+
+		const sasOptions = {
+			services: AccountSASServices.parse("b").toString(),          // blobs
+			resourceTypes: AccountSASResourceTypes.parse("sco").toString(), // service, container, object
+			permissions: AccountSASPermissions.parse("rwdlacupi"),          // permissions
+			protocol: SASProtocol.Https,
+			startsOn: new Date(),
+			expiresOn: new Date(new Date().valueOf() + (30 * 60 * 1000)),   // 30 minutes
+		};
+
+		const sasToken = generateAccountSASQueryParameters(
+			sasOptions,
 			sharedKeyCredential
 		).toString();
 
 		try {
 			for (let i = 0; i < templates.length; i++) {
```

Comment:
These templates depend on each other/ we need to upload them in a particular order? I think we can use promises here to upload concurrently

Reply:
there will be only a single template in majority of cases